src/aspire/utils/resolution_estimation.py

```diff
@@ -0,0 +1,394 @@
+"""
+This module contains code for estimating resolution achieved by reconstructions.
+"""
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from aspire.nufft import nufft
+from aspire.numeric import fft
+from aspire.utils import grid_2d
+
+logger = logging.getLogger(__name__)
+
+# FourierCorrelation holds a single implementation for both FSC and
+# FRC based on dimension `dim`.
+
+
+class FourierCorrelation:
+    r"""
+    Compute the Fourier correlations between two arrays.
+
+    Underlying data (images/volumes) are assumed to be well aligned.
+
+    The Fourier correlation is defined as:
+
+    .. math::
+
+       c(i) = \frac{ \operatorname{Re}( \sum_i{ \mathcal{F}_1(i) * {\mathcal{F}^{*}_2(i) } } ) }{\
+         \sqrt{ \sum_i { | \mathcal{F}_1(i) |^2 } * \sum_i{| \mathcal{F}^{*}_2}(i) |^2 } }
+.
+    """
+
+    def __init__(self, a, b, pixel_size=1, cutoff=0.143, eps=1e-4, method="fft"):
+        """
+        :param a: Input array a, shape(..., *dim).
+        :param b: Input array b, shape(..., *dim).
+        :param pixel_size: Pixel size in Angstrom.
+            Defaults to 1.
+        :param cutoff: Cutoff value, traditionally `.143`.
+        :param eps: Epsilon past boundary values, defaults 1e-4.
```

Comment:
The matlab implementation used this eps to avoid potential shell boundary numerical equality issues.  We don't need it other than for that legacy.  Since it wasn't clear to you what it was for, I assume it isn't important and I just removed it.